lib/SimpleSAML/XML/Parser.php

```diff
@@ -41,7 +41,10 @@ public static function fromSimpleXMLElement(\SimpleXMLElement $element)
         /* Create a new parser with the xml document where the namespace definitions
          * are added.
          */
-        $parser = new Parser($element->asXML());
+
+        /** @var string $xml */
+        $xml = $element->asXML();
+        $parser = new Parser($xml);
         return $parser;
```

Comment:
At some point, we end up with some very ugly code due to psalm 😞 

This should be `return new Parser($element->asXML());`...

Reply:
I may be able to fix that using a `/** @psalm-return Parser */` on the method's docblock.. I'll give it a try

Reply:
Good one! I was completely unaware of that option...

Reply:
I've also learned about this a couple of days ago..